examples/building.rs

```diff
@@ -211,10 +213,308 @@ impl Config for Game {
                             let place_at = location.get_in_direction(face);
                             if let Some(stack) = client.held_item() {
                                 if let Some(block_kind) = stack.item.to_block_kind() {
-                                    world.chunks.set_block_state(
-                                        place_at,
-                                        BlockState::from_kind(block_kind),
-                                    );
+                                    let state = match block_kind {
+                                        // Torches
+                                        BlockKind::Torch => face_wall_block(
+                                            face,
+                                            BlockState::TORCH,
+                                            BlockState::WALL_TORCH,
+                                        ),
+                                        BlockKind::RedstoneTorch => face_wall_block(
+                                            face,
+                                            BlockState::REDSTONE_TORCH,
+                                            BlockState::REDSTONE_WALL_TORCH,
+                                        ),
+                                        BlockKind::SoulTorch => face_wall_block(
+                                            face,
+                                            BlockState::SOUL_TORCH,
+                                            BlockState::SOUL_WALL_TORCH,
+                                        ),
```

Comment:
There's got to be a better way to do this than having this massive match right? This is an example, they're supposed to be simple enough for someone to know what they do at a glance. Do we have extracted metadata for this? Can we provide it as a utility in valence?

Reply:
Yeah, its getting pretty long, and there is still missing blocks. I can't seem to find any metadata on the normal <-> wall variants, but there should be something somewhere in the java code. But I mabye found something inside `blocks.java`:
```java
public static final Block PLAYER_HEAD = Blocks.register("player_head", new PlayerSkullBlock(AbstractBlock.Settings.of(Material.DECORATION).strength(1.0f)));
public static final Block PLAYER_WALL_HEAD = Blocks.register("player_wall_head", new WallPlayerSkullBlock(AbstractBlock.Settings.of(Material.DECORATION).strength(1.0f).dropsLike(PLAYER_HEAD)));
```
The important part is the `.dropsLike(PLAYER_HEAD)`, this shows atleast that they are connected, so this could mabye be used?

I thought about moving it into `build/blocks.rs` or should it have it's own file inside Valence?

Reply:
I have found the place where the metadata is, it's stored inside `item.java`, e.g:
```java
public static final Item TORCH = Items.register(new WallStandingBlockItem(Blocks.TORCH, Blocks.WALL_TORCH, new Item.Settings().group(ItemGroup.DECORATIONS)));
```
Here the constructor takes in two blocks, the `standingBlock` and `wallBlock` so I just need to extract these two (It's exactly identical for the other standing <-> wall blocks).